### PR TITLE
Add option --no-initdb to create and init for easier importing of other CKANs

### PR DESCRIPTION
**Pull request by JackMc**
Addresses #323

### CODE REVIEW

**Comment by wardi** [comment]
This new switch should force --image-only

**Comment by wardi** [comment]
or maybe require that the user also specifies --image-only? That might make it clearer

```
datacats create [-bn] [-s NAME] [--address=IP] [--syslog] [--ckan=CKAN_VERSION]
                  [--no-datapusher] [--site-url SITE_URL] [-i [--no-init-db]]
                  ENVIRONMENT_DIR [PORT]
```

**Comment by deniszgonjanin** [comment]
The command works beautifully, but running paster commands afterwards gives me:
```
FATA[0000] Error response from daemon: Could not get container for datacats_datapusher_bcgov_primary
```

Is there something in datacats that relies on datapusher running when running a datacats shell/paster command is ran?

**Comment by JackMc** [comment]
I'm guessing what's going on is that data pusher is failing cause the database is broken. Try with --no-datapusher.

**Comment by deniszgonjanin** [comment]
yes, it works well when env is created without datapusher. Unfortunately we can't have a --no-datapusher command on any command except create, because datapusher is automatically on when the directory is there. So `datacats init --no-init-db` will break for most datacats instances.

Any ideas on how we can easily have `datacats init --no-init-db --no-datapusher` work? I think the --no-init-db command is really more useful on existing or pre-packages datacats environments

**Comment by JackMc** [comment]
We can make the detection logic more complicated for data pusher (i.e. have a flag in the Env class)

**Comment by JackMc** [comment]
@wardi It already does force --image-only and --no-sysadmin

File: datacats/cli/create.py [hunk 1]
```diff
@@ -25,16 +25,18 @@ def create(opts):
...
+                  [--no-datapusher] [--site-url SITE_URL] [--no-init-db] ENVIRONMENT_DIR
+                  [PORT]
 
 Options:
   --address=IP            Address to listen on (Linux-only)
   --ckan=CKAN_VERSION     Use CKAN version CKAN_VERSION [default: 2.3]
   -b --bare               Bare CKAN site with no example extension
   -i --image-only         Create the environment but don't start containers
+  --no-datapusher         Don't install/enable ckanext-datapusher
+  --no-init-db            Don't initialize the database. Useful for importing CKANs.
```
Comment (wardi) [inline datacats/cli/create.py]:
```
--no-init-db            Advanced use: database must be manually
                        initialized before site will start. Implies
                        --no-datapusher and --image-only options.
```

File: datacats/cli/create.py [hunk 2]
```diff
@@ -55,12 +57,16 @@ def create(opts):
...
+        init_db=not opts['--no-init-db'],
         )
 
 
 def create_environment(environment_dir, port, ckan_version, create_skin,
         site_name, start_web, create_sysadmin, address, log_syslog=False,
-        datapusher=True, quiet=False, site_url=None):
+        datapusher=True, quiet=False, site_url=None, init_db=True):
+    if not init_db:
+        print 'Since the database will not be initialized, we will not copy datapusher.'
```
Comment (wardi) [inline datacats/cli/create.py]: not necessary if explained in comment as above

**Pull request closed without merging**